functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py

```diff
@@ -541,3 +543,56 @@ def test012_check_cloned_vm(self):
 
         with self.assertRaises(HTTPError) as e:
              self.api.cloudapi.machines.rollbackSnapshot(machineId=cloned_vm_id, epoch=snapshotEpoch)
+
+        self.lg('%s ENDED' % self._testID)
+
+    def test014_check_disk_iops_limit(self):
+        """ OVC-046
+        *Test case for checking cloned VM ip, portforwards and credentials*
+        **Test Scenario:**
+        #. Create virtual machine (VM1), should succeed.
+        #. Attach data disk (DD1) to VM1 and set MaxIOPS to iops1.
+        #. Run fio on DD1, iops should be less than iops1.
+        #. Change DD1's MaxIOPS limit to iops2 which is double iops1.
+        #. Run fio on DD1 again, iops should be between iops1 and iops2.
+        """
+        self.lg('%s STARTED' % self._testID)
+
+        def get_iops(vm_connection, run_name):
+            fio_cmd = "fio --ioengine=libaio --group_reporting --direct=1 --filename=/dev/vdb "\
+                      "--runtime=30 --readwrite=rw --rwmixwrite=5 --size=500M --name=test{0} "\
+                      "--output={0}".format(run_name)
+            vm1_conn.sudo(fio_cmd)
+            out1 = vm1_conn.sudo("cat %s | grep -o 'iops=[0-9]\{1,\}' | cut -d '=' -f 2" % run_name)
+            iops_list = out1.split('\r\n')
+            return iops_list
+
+        self.lg("Create virtual machine (VM1), should succeed.")
+        images = self.api.cloudapi.images.list()
+        image_id = [i['id'] for i in images if 'Ubuntu' in i['name']][0]
```

Comment:
in case the environment has no ubuntu image this line will produce out of index error 
need to check iimage_id  first 
for example 
```python
image_id = [i['id'] for i in images if 'Ubuntu' in i['name']]
if not image_id:
    self.skip('message here')

```